api/src/test/java/com/capitalone/dashboard/service/FeatureBranchServiceImplTest.java

```diff
@@ -0,0 +1,155 @@
+
+    private static FeatureBranch featureBranchOne;
+    private static FeatureBranch featureBranchTwo;
+    private static FeatureBranch featureBranchThree;
+    private static FeatureBranch featureBranchFour;
+
+    @Mock
+    private FeatureBranchRepository featureBranchRepository;
+    
+    @InjectMocks
+    private FeatureBranchServiceImpl featureBranchService;
+    
+
+    @Before
+    public void setup() {
+
+        featureBranchOne = new FeatureBranch();
+        featureBranchOne.setName("F1");
+        featureBranchOne.setCommitIdFirstCommit("G001_FC");
+        featureBranchOne.setCommitIdThatTriggeredDeploy("G011_TB");
+        featureBranchOne.setGitRepoUrl("api.github.com/repo/commits/G001-G011");
+        featureBranchOne.setFirstCommitTimeStamp(1499153944L);      //2017/07/04 :: 7:39:04
+        featureBranchOne.setDeployTimeStamp(1499253944L);           // 2017/07/05 :: 11:25:44
+        
+        featureBranchTwo = new FeatureBranch();
+        featureBranchTwo.setName("F2");
+        featureBranchTwo.setCommitIdFirstCommit("G002_FC");
+        featureBranchTwo.setCommitIdThatTriggeredDeploy("G022_TB");
+        featureBranchTwo.setGitRepoUrl("api.github.com/repo/commits/G002-G022");
+        featureBranchTwo.setFirstCommitTimeStamp(1499353944L);      // 2017/07/06 :: 15:12:24
+        featureBranchTwo.setDeployTimeStamp(1499453944L);           //2017/07/07 :: 18:59:04
+
+        featureBranchThree = new FeatureBranch();
+        featureBranchThree.setName("F3");
+        featureBranchThree.setCommitIdFirstCommit("G003_FC");
+        featureBranchThree.setCommitIdThatTriggeredDeploy("G033_TB");
+        featureBranchThree.setGitRepoUrl("api.github.com/repo/commits/G003-G033");
+        featureBranchThree.setFirstCommitTimeStamp(1499553944L);    // 2017/07/08 22:45:54
+        featureBranchThree.setDeployTimeStamp(1499653944L);         //2017/07/10 02:32:24   
+
+        featureBranchService.save(featureBranchOne);
+        featureBranchService.save(featureBranchTwo);
+        featureBranchService.save(featureBranchThree);
+
+    }
+
+    @After
+    public void cleanup() {
+     
+     featureBranchOne = null;
+     featureBranchTwo = null;
+     featureBranchThree = null;
+     featureBranchRepository.deleteAll();
+    }
+
+    private List<FeatureBranch> getFeatureBranchList()
+    {
+        List<FeatureBranch> featureBranchList = new ArrayList();
+
+        featureBranchList.add(featureBranchOne);
+        featureBranchList.add(featureBranchTwo);
+        featureBranchList.add(featureBranchThree);
+
+        return featureBranchList;
+    }
+
+    @Test
+    public void validateSave() {
+
+        List<FeatureBranch> featureBranchList = getFeatureBranchList();
+
+        when(featureBranchRepository.findAll()).thenReturn(featureBranchList);
+
+        assertEquals("Save function validation for the FeatureBranchRepository has failed",
+                featureBranchService.findAll().size(),3);
```

Comment:
Did you mean? https://stackoverflow.com/questions/11598977/get-size-of-an-iterable-in-java#11599011